app/models/vote.rb

```diff
@@ -0,0 +1,17 @@
+class Vote < ApplicationRecord
+  belongs_to :voter,
+             class_name:  :User,
+             foreign_key: :user_id
+
+  belongs_to :voteable,
+              polymorphic: true
+
+  validates_uniqueness_of :user_id,
+                          scope: [:voteable_id, :voteable_type]
+
+  validates_numericality_of :value, {
+                                      only_integer:             true,
+                                      greater_than_or_equal_to: -1,
+                                      less_than_or_equal_to:    1
+                                    }
```

Comment:
not super critical or anything, but both of these validations could've just used the `validates` method like so:

```ruby
class Vote < ApplicationRecord
  validates :user_id,
            uniqueness: { scope: [:voteable_id, :voteable_type] }

  validates :value,
            numericality: {
              only_integer:             true,
              greater_than_or_equal_to: -1,
              less_than_or_equal_to:    1
            }
end
```

Saves some with indentation.

Another thing is that the numericality validation could be replaced with an inclusion. something like:

```ruby
validates :value, inclusion: { in: [1, 0, -1] }
```